tests/ecmp/test_ecmp_balance.py

```diff
@@ -0,0 +1,719 @@
+
+    # Find interfaces with tx_ok larger than PACKET_COUNT
+    interfaces_with_high_tx = []
+    for iface, stats in portstat_out.items():
+        tx_ok = int(stats["tx_ok"].replace(",", ""))
+        if tx_ok >= PACKET_COUNT:
+            interfaces_with_high_tx.append(iface)
+
+    if interfaces_with_high_tx:
+        logger.info(
+            f"Interfaces with tx_ok >= {PACKET_COUNT}: {interfaces_with_high_tx}"
+        )
+    else:
+        logger.warning(f"No interfaces found with tx_ok >= {PACKET_COUNT}")
+
+    # Assert that at least one interface forwarded enough packets
+    pytest_assert(
+        len(interfaces_with_high_tx) > 0,
+        f"No interfaces found with tx_ok >= {PACKET_COUNT}",
+    )
+
+    # Assert that PTF matched the expected number of packets
+    pytest_assert(
+        match_cnt >= PACKET_COUNT,
+        f"PTF matched {match_cnt} packets, which is less than the expected {PACKET_COUNT}",
+    )
+    # Sort interfaces by tx_ok count in descending order
+    sorted_interfaces = sorted(
+        interfaces_with_high_tx,
+        key=lambda iface: int(portstat_out[iface]["tx_ok"].replace(",", "")),
+        reverse=True,
+    )
+
+    # Return the interface with highest tx_ok count if available
+    return sorted_interfaces[0] if sorted_interfaces else None
+
+
+def save_test_results(duthost, test_results, ip_version, with_ecmpoffset=False):
+    if with_ecmpoffset:
+        file_name = (
+            f"logs/{duthost.hostname}_{ip_version}_hash_tuples_test_results_ecmp.json"
+        )
+        file_name_formatted = f"logs/{duthost.hostname}_{ip_version}_hash_tuples_test_results_formatted_ecmp.json"
+    else:
+        file_name = (
+            f"logs/{duthost.hostname}_{ip_version}_hash_tuples_test_results.json"
+        )
+        file_name_formatted = f"logs/{duthost.hostname}_{ip_version}_hash_tuples_test_results_formatted.json"
+    try:
+        local_results_file = os.path.join(os.getcwd(), file_name)
+        with open(local_results_file, "w") as f:
+            json.dump(test_results, f, indent=4)
+        logger.info(f"Test results saved to {local_results_file}")
+    except Exception as e:
+        # Format the test results to organize by interface
+        logger.error(f"Failed to save test results to file: {e}")
+    format_results = {}
+
+    # Process each pattern
+    for pattern in test_results:
+        format_results[pattern] = {}
+
+        # Group by interface
+        for case, data in test_results[pattern].items():
+            interface = data["out_interface"]
+            if not interface:
+                continue
+
+            if interface not in format_results[pattern]:
+                format_results[pattern][interface] = {}
+                format_results[pattern][interface]["tuples"] = {}
+                format_results[pattern][interface]["count"] = 0
+
+            # Add this tuple to the interface's list
+            tuple_key = f"tuples_{case}"
```

Comment:
hi @ZhaohuiS  just one question, what's the case here? Stands for 5 tuples? sip/dip/sport/dport/protocol?

Reply:
@StormLiangMS  yes, it's a combination of 5 tuples, test results is like the following sample:


```
{
    "pattern_1": {
        "1": {
            "5_tuples": "src=20.0.0.1 dst=194.50.16.1 sport=100 dport=80 proto=17",
            "out_interface": "Ethernet116",
            "packets_count": 100
        },
```